contracts/BackedAutoFeeTokenImplementation.sol

```diff
@@ -328,16 +357,24 @@ contract BackedAutoFeeTokenImplementation is BackedTokenImplementation {
     function updateMultiplierValue(
         uint256 newMultiplier,
         uint256 oldMultiplier
-    ) external updateMultiplier {
-        require(
-            _msgSender() == multiplierUpdater,
-            "BackedToken: Only multiplier updater"
-        );
-        require(
-            multiplier == oldMultiplier,
-            "BackedToken: Multiplier changed in the meantime"
-        );
-        _updateMultiplier(newMultiplier);
+    ) public onlyMultiplierUpdater onlyUpdatedMultiplier(oldMultiplier) updateMultiplier {
+        _updateMultiplier(newMultiplier, multiplierNonce + 1);
+    }
+
+    /**
+     * @dev Function to change the contract multiplier with nonce, only if oldMultiplier did not change in the meantime. Allowed only for owner
+     *
+     * Emits a { MultiplierChanged } event
+     *
+     * @param newMultiplier New multiplier value
+     * @param newMultiplierNonce New multplier nonce
+     */
+    function updateMultiplierWithNonce(
+        uint256 newMultiplier,
+        uint256 oldMultiplier,
+        uint256 newMultiplierNonce
+    ) external onlyMultiplierUpdater onlyUpdatedMultiplier(oldMultiplier) onlyNewerMultiplierNonce(newMultiplierNonce) updateMultiplier {

```

Comment:
updateMultiplier can change nonce after check in onlyNewerMultiplierNonce is executed. Is it expected behaviour?